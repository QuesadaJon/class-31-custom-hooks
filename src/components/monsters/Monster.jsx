import React from 'react';
import PropTypes from 'prop-types';

const Monster = ({ name, image }) => (
  <figure>
    <p>{name}</p>
    <img src={image} alt={name} />
  </figure>
  
);

Monster.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Monster;
