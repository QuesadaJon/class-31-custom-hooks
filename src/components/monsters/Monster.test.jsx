import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Monster from './Monster';

describe('Monster component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Monster 
          name="Fredrick"
          image="http//image.com"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
})
;
