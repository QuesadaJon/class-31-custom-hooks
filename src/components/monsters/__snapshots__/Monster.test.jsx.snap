// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monster component matches a snapshot 1`] = `
<DocumentFragment>
  <figure>
    <p>
      Fredrick
    </p>
    <img
      alt="Fredrick"
      src="http//image.com"
    />
  </figure>
</DocumentFragment>
`;
